DependencyInjection/Factory/Resolver/AwsS3ResolverFactory.php

```diff
@@ -15,11 +15,15 @@ class AwsS3ResolverFactory implements ResolverFactoryInterface
      */
     public function create(ContainerBuilder $container, $resolverName, array $config)
     {
+        $awsS3ClientId = 'liip_imagine.cache.resolver.'.$resolverName.'.client';
         $awsS3ClientDefinition = new Definition('Aws\S3\S3Client');
-        $awsS3ClientDefinition->setFactoryClass('Aws\S3\S3Client');
-        $awsS3ClientDefinition->setFactoryMethod('factory');
+        if (method_exists($awsS3ClientDefinition, 'setFactory')) {
```

Comment:
okay, could add a comment to `else` statement that it can be remove when we drop support of symfony <2.7